go-kit/metrics/provider/librato/librato.go

```diff
@@ -238,6 +256,17 @@ type attributes struct {
 	Aggregate bool `json:"aggregate,omitempty"`
 }
 
+// reportWithRetry the metrics to the url, every interval, with max retries.
+func (p *Provider) reportWithRetry(u *url.URL, interval time.Duration) {
+	for i := p.numRetries; i > 0; i-- {
+		err := p.report(u, interval)
+		if err == nil {
+			return
+		}
+		p.errorHandler(err)
```

Comment:
`errorHandler` may be `nil` and and IMO the error passes should have a way to indicate if it's temporary() or not.